### DISCUSSION TITLE
[Socks] Add support for v4 and v5 socks proxy

### DISCUSSION THREAD

**Discussion by abhinavsingh** [abhinavsingh/proxy.py#793] [Ideas]
**Describe the solution you'd like**
Socks4 / Socks5 support

**Comment by thiagobenine** [comment]
Any update on this? This is a really interesting feature.

**Comment by abhinavsingh** [comment]
@thiagobenine IIRC, I wrote an initial module for `Socksv4`.  See https://github.com/abhinavsingh/proxy.py/tree/develop/proxy/socks. It was just never tried/released in the wild.  Having said that, I think it still need some work.  For, `Socksv5`, we'll need more time or only implement parts of it which are of use to `proxy.py` users.

I fear at this point there is no way for users to try `socksv4`, even though some code exists.    Bigger problem is, I don't have a real world socks use-case at my end :(. Neither do I see bandwidth on my plate to implement this anytime sooner.  Considering all this in context of `proxy.py`, supporting `socks` for all versions of python 3.6-3.10, having decent test coverage, will take considerable time and effort.

I'll spend sometime over the weekend and see if we can kick start `socks4` somehow, so that other contributors/users can atleast try existing code and possibly take it forward as necessary.
